go/billing/api.py

```diff
@@ -247,6 +250,32 @@ def create_transaction_interaction(self, cursor, account_number,
         session_len_credits = MessageCost.calculate_session_length_credit_cost(
             session_len_cost, markup_percent)
 
+        query = """SELECT credit_balance, last_topup_balance
+                   FROM billing_account
+                   WHERE account_number = %(account_number)s"""
+
+        params = {'account_number': account_number}
+        cursor = yield cursor.execute(query, params)
+        result = yield cursor.fetchone()
+
+        if result is None:
+            raise BillingError(
+                "Unable to find billing account %s while checking"
+                " credit balance. Message was %s to/from tag pool %s." % (
+                    account_number, message_direction, tag_pool_name))
+
+        last_topup_balance = result.get('last_topup_balance')
+        credit_balance = result.get('credit_balance')
+
+        def credit_percent():
+            return self._ceil_percent(credit_balance, last_topup_balance)
+
+        # If the message is outbound and limit is reached, don't charge
+        if (app_settings.ENABLE_LOW_CREDIT_CUTOFF and last_topup_balance and
+                message_direction == MESSAGE_DIRECTION_OUTBOUND):
+            if credit_percent() < self._notification_mapping[0]:
+                defer.returnValue({'credit_cutoff_reached': True})
```

Comment:
Perhaps we should also explicitly record whether a transaction was created in the result returned?

I was vaguely thinking of changing the result to something like:
```
{
    transaction: { ... },  /* or transaction: null if there isn't one */
    credit_cutoff_reached: true
}
```

That would require adjusting the dispatcher at the same time though. If we think it's worth it, perhaps it's something for another PR?